eureka/S5_lightcurve_fitting/models.py

```diff
@@ -152,7 +151,11 @@ def plot(self, time, components=False, ax=None, draw=False, color='blue', zorder
         label = self.fitter
         if self.name!='New Model':
             label += ': '+self.name
-        ax.plot(self.time, self.eval(**kwargs), '-', label=label, color=color, zorder=zorder)
+        if share:
```

Comment:
You could reduce duplication here by doing something like:
```
flux = self.eval(**kwargs)
if share:
    onelc=len.....
    flux = flux[onelc*......
ax.plot(self.time, flux, .........
```

Reply:
I think the nchannel=1 thing fixed it anyway